client/src/app/core/services/csv-export.service.ts

```diff
@@ -0,0 +1,124 @@
+    ): void {
+        const all_lines = []; // Array of arrays of entries
+        const used_columns = []; // mapped properties to be included
+
+        // initial array of usable text separators. The first character not used
+        // in any text data or as column separator will be used as text separator
+        let ts_list = ['"', "'", '`', '/', '\\', ';', '.', ','];
+
+        if (lineSeparator === columnSeparator) {
+            throw new Error('lineseparator and columnseparator must differ from each other');
+        }
+
+        ts_list = this.checkCsvTextSafety(lineSeparator, ts_list);
+        ts_list = this.checkCsvTextSafety(columnSeparator, ts_list);
+
+        // create header data
+        const header = [];
+        columns.forEach(column => {
+            const rawLabel: string = column.label ? column.label : (column.property as string);
+            const colLabel = this.capitalizeTranslate(rawLabel);
+            ts_list = this.checkCsvTextSafety(colLabel, ts_list);
+            header.push(colLabel);
+            used_columns.push(column.property);
+        });
+        all_lines.push(header);
+        // create lines
+        data.forEach(item => {
+            const line = [];
+            used_columns.forEach(property => {
+                ts_list = this.checkCsvTextSafety(item[property], ts_list);
+                line.push(item[property]);
+            });
+            all_lines.push(line);
+        });
+
+        // assemble lines, putting text separator in place
+        const final_lines = [];
+        if (!ts_list.length) {
+            throw new Error('no usable text separator left for valid csv text');
+        }
+        all_lines.forEach(function(line: any[]): void {
+            const newline = [];
+            line.forEach(function(item: any): void {
+                if (typeof item === 'number') {
+                    newline.push(item);
+                } else if (item === null || item === undefined || item === '') {
+                    newline.push('');
+                } else if (item === true) {
+                    newline.push('1');
+                } else if (item === false) {
+                    newline.push('0');
+                } else {
+                    const newstring = ts_list[0] + item + ts_list[0];
+                    newline.push(newstring);
+                }
+            });
+            final_lines.push(newline.join(columnSeparator));
+        });
+
+        this.exporter.saveFile(final_lines.join(lineSeparator), filename);
+    }
+
+    /**
+     * Checks if a given input contains any of the characters defined in a list
+     * used for textseparators. The list is then returned without the 'special'
+     * characters, as they may not be used as text separator in this csv.
+     * @param input any input to be sent to CSV
+     * @param ts_list The list of special characters to check.
+     */
+    public checkCsvTextSafety(input: any, ts_list: string[]): string[] {
```

Comment:
type of input: `string`. Cast above if accessing `item[property]`.

The complete function may be cut down to `return tsList.filter(char => input.indexOf(char) < 0);`. This needs testing but should work.

Reply:
At this type I should not be required to rely on it being a string: For later parsing, I will need to escape texts into a textseparator (which I'm finding out at this stage). If I cast is to a string above, I'll lose the ability to check for numbers or booleans further in the line assembly